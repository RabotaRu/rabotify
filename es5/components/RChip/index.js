'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _RChip = require('./RChip.vue');

var _RChip2 = _interopRequireDefault(_RChip);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* istanbul ignore next */
_RChip2.default.install = function install(Vue) {
  Vue.component('r-chip', _RChip2.default);
};

exports.default = _RChip2.default;