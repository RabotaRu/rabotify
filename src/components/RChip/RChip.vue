<script>
  import Colorable from '../../mixins/colorable';
  import Toggleable from '../../mixins/toggleable';

  // Icons
  import MdClose from '@rabota/md-svg-vue/dist/navigation/MdClose.vue';

  // Components
  import RBtn from '../RBtn/RBtn.vue';

  export default {
    name: 'r-chip',

    mixins: [
      Colorable,
      Toggleable
    ],

    components: {
      MdClose,
      RBtn
    },

    props: {
      close: Boolean,
      disabled: Boolean,
      label: Boolean,
      outline: Boolean,
      round: Boolean,
      // Used for selects/tagging
      selected: Boolean,
      small: Boolean,
      textColor: String,
      value: {
        type: Boolean,
        default: true
      }
    },

    computed: {
      classes () {
        const classes = this.addBackgroundColorClassChecks({
          'r-chip_disabled': this.disabled,
          'r-chip_selected': this.selected,
          'r-chip_label': this.label,
          'r-chip_outline': this.outline,
          'r-chip_round': this.round,
          'r-chip_small': this.small,
          'r-chip_removable': this.close,
          'r-chip_default': !this.color
        });

        return (this.textColor || this.outline)
          ? this.addTextColorClassChecks(classes, this.textColor ? 'textColor' : 'color')
          : classes;
      }
    },

    methods: {
      genClose (h) {
        const data = {
          staticClass: 'r-chip__close',
          props: {
            icon: true,
            flat: true,
            ripple: true
          },
          on: {
            click: e => {
              e.stopPropagation();

              this.$emit('input', false);
            }
          }
        };

        return h('r-btn', data, [ h('md-close') ]);
      },
      genContent (h) {
        const children = [ this.$slots.default ];

        this.close && children.push(this.genClose(h));

        return h('span', {
          staticClass: 'r-chip__content'
        }, children);
      }
    },

    render (h) {
      const data = {
        staticClass: 'r-chip',
        'class': this.classes,
        attrs: { tabindex: this.disabled ? -1 : 0 },
        directives: [{
          name: 'show',
          value: this.isActive
        }],
        on: this.$listeners
      };

      return h('span', data, [ this.genContent(h) ]);
    }
  };
</script>

<style lang="scss">
  @import "../../styles/components/_chip.scss";
</style>
