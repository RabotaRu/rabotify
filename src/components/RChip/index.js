import RChip from './RChip.vue';

/* istanbul ignore next */
RChip.install = function install (Vue) {
  Vue.component('r-chip', RChip);
};

export default RChip;
