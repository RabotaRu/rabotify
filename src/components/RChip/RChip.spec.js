import RChip from '@components/RChip';
import { test } from '@util/testing';

test('RChip.vue', ({ mount, compileToFunctions }) => {
  it('should have a r-chip class', () => {
    const wrapper = mount(RChip);

    expect(wrapper.hasClass('r-chip')).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should be removable', () => {
    const wrapper = mount(RChip, {
      propsData: { close: true }
    });

    const close = wrapper.find('.r-chip__close')[0];

    const input = jest.fn(value => wrapper.setProps({ value }));
    wrapper.vm.$on('input', input);

    expect(wrapper.html()).toMatchSnapshot();

    close.trigger('click');
    expect(input).toBeCalledWith(false);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render a colored chip', () => {
    const wrapper = mount(RChip, {
      propsData: {
        color: 'secondary',
        textColor: 'accent'
      }
    });

    expect(wrapper.element.classList).toContain('secondary');
    expect(wrapper.element.classList).toContain('text_accent');
  });

  it('should render a disabled chip', () => {
    const wrapper = mount(RChip, {
      propsData: {
        disabled: true
      }
    });

    expect(wrapper.element.classList).toContain('r-chip_disabled');
  });

  it('should render a colored outline chip', () => {
    const wrapper = mount(RChip, {
      propsData: {
        outline: true,
        color: 'primary'
      }
    });

    expect(wrapper.element.classList).toContain('primary');
    expect(wrapper.element.classList).toContain('text_primary');
  });

  it('should render a colored outline chip with text color', () => {
    const wrapper = mount(RChip, {
      propsData: {
        outline: true,
        color: 'accent',
        textColor: 'info'
      }
    });

    expect(wrapper.element.classList).toContain('accent');
    expect(wrapper.element.classList).toContain('text_info');
  });
});
