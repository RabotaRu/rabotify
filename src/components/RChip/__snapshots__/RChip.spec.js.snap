// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RChip.vue should be removable 1`] = `

<span tabindex="0"
      class="r-chip r-chip_removable r-chip_default"
>
  <span class="r-chip__content">
    <button aria-label="uniqlabel"
            type="button"
            class="r-chip__close r-btn r-btn_flat r-btn_icon r-btn_medium"
            data-ripple="true"
    >
      <svg width="24"
           height="24"
           viewbox="0 0 24 24"
           xmlns="http://www.w3.org/2000/svg"
           class="icon md-icon md-close"
      >
        <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
        </path>
      </svg>
    </button>
  </span>
</span>

`;

exports[`RChip.vue should be removable 2`] = `

<span tabindex="0"
      class="r-chip r-chip_removable r-chip_default"
      style="display: none;"
>
  <span class="r-chip__content">
    <button aria-label="uniqlabel"
            type="button"
            class="r-chip__close r-btn r-btn_flat r-btn_icon r-btn_medium"
            data-ripple="true"
    >
      <svg width="24"
           height="24"
           viewbox="0 0 24 24"
           xmlns="http://www.w3.org/2000/svg"
           class="icon md-icon md-close"
      >
        <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
        </path>
      </svg>
    </button>
  </span>
</span>

`;

exports[`RChip.vue should have a r-chip class 1`] = `

<span tabindex="0"
      class="r-chip r-chip_default"
>
  <span class="r-chip__content">
  </span>
</span>

`;
